agilent_infiniium_scope.py

```diff
@@ -38,29 +38,29 @@
 from twisted.internet.defer import inlineCallbacks, returnValue
 import labrad.units as U
 from struct import unpack, calcsize
-
-import numpy, re
+import numpy as np
 
 COUPLINGS = ['AC', 'DC', 'GND']
 TRIG_CHANNELS = ['AUX','CH1','CH2','CH3','CH4','LINE']
 VERT_DIVISIONS = 8.0
 HORZ_DIVISIONS = 10.0
 SCALES = []
 
+
 class AgilentDSO91304AServer(GPIBManagedServer):
     name = 'Agilent Infiniium Oscilloscope'
     deviceName = ['Agilent Technologies DSO91304A', 'KEYSIGHT TECHNOLOGIES DSO90804A']
-        
+
     @setting(11, returns=[])
     def reset(self, c):
         dev = self.selectedDevice(c)
-        yield dev.write('*RST')
+        yield dev.query('*RST;*OPC?')
```

Comment:
It is not related, but an improvement to the server after talking to Keysight support.  The *OPC? query confirms that the scope has finished parsing commands.  It does not mean that the scope has actually finished executing the previous command.  Keysight support recommended adding in some *OPC? queries in order to give the scope time to execute some of the longer commands.